classes/actions/ShoppingfeedOrderImportActions.php

```diff
@@ -717,12 +717,27 @@ public function validateOrder()
 
             return false;
         }
+
+        $transactionId = $apiOrder->getReference();
+        $paymentMethod = Tools::strtolower($apiOrder->getChannel()->getName());
+
+        $this->specificRulesManager->applyRules(
+            'onValidateOrder',
+            [
+                'apiOrder' => $apiOrder,
+                'orderData' => &$this->conveyor['orderData'],
+                'cart' => &$cart,
+               'paymentMethod' => &$paymentMethod,
+               'transactionId' => &$transactionId,
+            ]
+        );
+
         try {
             $paymentModule->validateOrder(
                 (int) $cart->id,
                 $this->initSfOrderState()->get()->id,
                 $amount_paid,
-                Tools::strtolower($apiOrder->getChannel()->getName()),
+                $paymentMethod,
                 null,
                 ['transaction_id' => $apiOrder->getReference()],
```

Comment:
`['transaction_id' => $apiOrder->getReference()],`
to replace by
`['transaction_id' => $transactionId],`